From 59c0f15db5507d20760225ee82b24083a564b923 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E7=8E=8B=E5=AD=90=E5=8D=BF?= <wang_ph@sjtu.edu.cn>
Date: Mon, 6 Jul 2026 22:16:04 +0800
Subject: [PATCH 5/5] =?UTF-8?q?=E5=8E=BB=E6=8E=89libc=E7=9A=84=E4=BF=AE?=
 =?UTF-8?q?=E6=94=B9?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 bsp/abstract-machine/src/libc.c | 31 -------------------------------
 1 file changed, 31 deletions(-)

diff --git a/bsp/abstract-machine/src/libc.c b/bsp/abstract-machine/src/libc.c
index 4905cf7cd..b57efe16f 100644
--- a/bsp/abstract-machine/src/libc.c
+++ b/bsp/abstract-machine/src/libc.c
@@ -1,53 +1,22 @@
 #include <klib.h>
 #include <rtthread.h>
 
-<<<<<<< HEAD
-char *strchr(const char *s, int c) {
-	int i = 0;
-  for ( ; s[i] != '\0'; ++i) {
-		if ((int)s[i] == c) return (char *)s + i;
-	}
-	if (c == (int)'\0') return (char *)s + i;
-	return NULL;
-}
-
-char *strrchr(const char *s, int c) {
-	char *res = NULL;
-	int i = 0;
-  for ( ; s[i] != '\0'; ++i) {
-		if ((int)s[i] == c) res = (char *)s + i;
-	}
-	if (c == (int)'\0') return (char *)s + i;
-	return res;
-=======
 __attribute__((weak)) char *strchr(const char *s, int c) {
   assert(0);
 }
 
 __attribute__((weak)) char *strrchr(const char *s, int c) {
   assert(0);
->>>>>>> 72ba231bb79db53f1c07920f2642ef20679faeff
 }
 
 __attribute__((weak)) char *strstr(const char *haystack, const char *needle) {
   return rt_strstr(haystack, needle);
 }
 
-<<<<<<< HEAD
-long strtol(const char *restrict nptr, char **restrict endptr, int base) {
-  //assert(0);
-	return 0;
-}
-
-char *strncat(char *restrict dst, const char *restrict src, size_t sz) {
-  //assert(0);
-	return NULL;
-=======
 __attribute__((weak)) long strtol(const char *restrict nptr, char **restrict endptr, int base) {
   assert(0);
 }
 
 __attribute__((weak)) char *strncat(char *restrict dst, const char *restrict src, size_t sz) {
   assert(0);
->>>>>>> 72ba231bb79db53f1c07920f2642ef20679faeff
 }
-- 
2.34.1

